returnn/tf/layers/basic.py

```diff
@@ -6598,8 +6598,34 @@ def transform_config_dict(cls, d, network, get_layer):
     d["_subnet"] = subnet
     # In case of non-template construction, this will trigger the non-template construction of our "output" sublayer.
     d["_output"] = subnet.construct_layer("output", parent_get_layer=get_layer)
-    d["_from"] = d.get("from", "data")  # cache this
-    d["from"] = []  # disable now. we should get them in the template construction when needed
+    d["from"] = d.get("from", "data")
```

Comment:
But this is exactly what should **not** be done here.
The sources **must** only be accessed dynamically during the template construction of the subnet to correctly resolve dependencies only on those sub layers which really need a particular source.

Reply:
I doubt that there is a problem on this part in the code. I think `transform_config_dict` is fine and should be left like it is.

Reply:
Yes, editing the from was not correct, and I was aware of that. I added this to show what would fix one of the test cases. But why is the other change wrong? The `transform_config_dict`of the Recurrent layer has exactly the same code to resolve the dependecies, so why should it be correct for a recurrent subnetwork layer but be incorrect for a normal subnetwork layer?

FYI:
the "from" change was used to make `test_attention_subnetwork_from_dependency` work, and
copying the code from the recurrent layer that iterates the sublayers was used to make `test_attention_subnetwork_base_dependency` work.

Reply:
Because that was the whole point of all the recent `SubnetworkLayer` changes, that the whole `SubnetworkLayer` itself does not directly have the dependencies, but just its sub layers have. This is important to be able to split it up e.g. for optimization (e.g. some sub layers are inside the loop, some are outside).

Reply:
But this is only related to the "from" change (which I understand why this was wrong).

The other code should have no effect on this, otherwise this would not work for the recurrent layer.

I will update the PR accordingly and then we can have a look again.

Reply:
Okay, then I am definitely not able to solve this